src/ert/shared/models/multiple_data_assimilation.py

```diff
@@ -71,13 +67,38 @@ async def run(self, evaluator_server_config: EvaluatorServerConfig) -> None:
         weights_to_run = enumerated_weights[
             min(self._simulation_arguments["start_iteration"], len(weights)) :
         ]
-        run_context = self.create_context(0, initialize_mask_from_arguments=True)
-        self.ert().sample_prior(run_context.sim_fs, run_context.active_realizations)
+        starting_iteration = self._simulation_arguments["start_iteration"]
+        case_format = self._simulation_arguments["target_case"]
+        storage_manager = self.ert().storage_manager
+        if starting_iteration > 0:
+            if case_format % starting_iteration not in storage_manager:
+                raise ErtRunError(
+                    f"Source case {case_format % starting_iteration} for iteration "
+                    f"{starting_iteration} does not exists in {storage_manager.cases}."
+                    " If you are attempting to restart ESMDA from a iteration other "
+                    "than 0, make sure the target case format is the same as for the "
+                    f"original run.(Current target case format: {case_format})"
+                )
+        else:
+            prior_fs = storage_manager.add_case(case_format % starting_iteration)
+            self.ert().sample_prior(
+                prior_fs,
+                [
```

Comment:
Feel free to split this out of the function call, at least I think that would look better.